terraform-aws-github-runner/modules/runners-instances/templates/user-data.sh

```diff
@@ -104,6 +104,63 @@ ${install_config_runner}
 retry sudo dnf groupinstall -y 'Development Tools'
 retry sudo dnf install -y "kernel-devel-uname-r == $(uname -r)" || true
 
+%{ if wiz_secrets_arn != null ~}
+# Install Wiz Sensor - a runtime security agent
+echo "Fetching Wiz secrets from AWS Secrets Manager"
+
+(
+    # Allow the script to continue even if the function fails, so that we gracefully handle
+    # wiz installation failures
+    set +e
+
+    # Function to get region from ARN
+    get_region_from_arn() {
+        local arn=$1
+        # Extract region from ARN (format: arn:aws:service:region:account:resource)
+        local region=$(echo "$arn" | cut -d':' -f4)
+        if [ -n "$region" ]; then
+            echo "$region"
+        else
+            echo ""
+        fi
+    }
+
+    SECRET_REGION=$(get_region_from_arn "${wiz_secrets_arn}")
+    if [ -z "$SECRET_REGION" ]; then
+        echo "Warning: Region is required in the Secrets Manager ARN. Skipping Wiz installation."
+        metric_report "linux_userdata.wiz_failure_arn_invalid" 1
+    else
+        WIZ_SECRET_RAW=$(retry aws secretsmanager get-secret-value --secret-id "${wiz_secrets_arn}" --region "$SECRET_REGION" --query 'SecretString' --output text)
```

Comment:
Do you need to grant this permission on Terraform?  I remember I needed to do that for Docker Hub secrets in https://github.com/pytorch-labs/pytorch-gha-infra/pull/544